src/vaadin-grid-data-provider-mixin.html

```diff
@@ -345,7 +345,9 @@
           filters: this._mapFilters(),
           parentItem: cache.parentItem
         };
-
+        if (this.__scrollingToPendingIndex && this._debounceIncreasePool) {
```

Comment:
This new `__scrollingToPendingIndex` property feels a bit hacky? Is there some other justification for adding it other than making that one data-provider test pass / should the increase pool debouncer not be flushed while the grid is scrolling to pending index?

I took a look at the failing test case and it seems to pass because of timing. If I wrap the last expect inside of a timeout of 1 second, it also fails on master due to a too large call count. I don't necessarily see it as a problem to increase the call count expectation to make the test pass with the new change in timing.

Reply:
Done